src/vsphere_cpi/spec/integration/nsxt_spec.rb

```diff
@@ -1334,49 +1592,89 @@ def initialize(name: "BOSH-CPI-test-#{SecureRandom.uuid}", id: SecureRandom.uuid
+        else
+          create_static_server_pool(pool_1.name)
+        end
+      end
+      let!(:server_pool_2) do
+        if use_policy_api?
+          create_lb_pool(pool_2)
+        else
+          create_dynamic_server_pool(pool_2.name, nsgroup_1)
+        end
+      end
       after do
-        delete_server_pool(server_pool_1)
-        delete_server_pool(server_pool_2)
-        delete_nsgroup(nsgroup_1)
+        if use_policy_api?
+          delete_lb_pool(pool_1)
+          delete_lb_pool(pool_2)
+          delete_policy_group(nsgroup_name_1)
+        else
+          delete_server_pool(server_pool_1)
+          delete_server_pool(server_pool_2)
+          delete_nsgroup(nsgroup_1)
+        end
       end
       it "removes all the VM's logical ports from all NSGroups linked to dynamic server pool" do
-        lport_ids = []
-        simple_vm_lifecycle(cpi, '', vm_type, network_spec) do |vm_id|
-          verify_ports(vm_id) do |lport|
-            expect(lport).not_to be_nil
-            lport_ids << lport.id
+        if use_policy_api?
+          # For policy API, just verify VM creation and deletion works
+          simple_vm_lifecycle(cpi, '', vm_type, network_spec) do |vm_id|
+            expect(vm_id).to_not be_nil
           end
-        end
-        expect(lport_ids.length).to eq(2)
-
-        lport_ids.each do |id|
-          grouping_object_svc = NSXT::ManagementPlaneApiGroupingObjectsNsGroupsApi.new(@manager_client)
-          nsgroups = grouping_object_svc.list_ns_groups.results.select do |nsgroup|
-            next unless nsgroup.members
-            nsgroup.members.find do |member|
-              member.target_type == 'LogicalPort' && member.target_property == 'id' && member.value == id
+        else
+          # For manager API, test the original behavior
+          lport_ids = []
+          simple_vm_lifecycle(cpi, '', vm_type, network_spec) do |vm_id|
+            verify_ports(vm_id) do |lport|
+              expect(lport).not_to be_nil
+              lport_ids << lport.id
             end
           end
+          expect(lport_ids.length).to eq(2)
+
+          lport_ids.each do |id|
+            grouping_object_svc = NSXT::ManagementPlaneApiGroupingObjectsNsGroupsApi.new(@manager_client)
+            nsgroups = grouping_object_svc.list_ns_groups.results.select do |nsgroup|
+              next unless nsgroup.members
+              nsgroup.members.find do |member|
+                member.target_type == 'LogicalPort' && member.target_property == 'id' && member.value == id
+              end
+            end
 
-          expect(nsgroups).to eq([])
+            expect(nsgroups).to eq([])
+          end
         end
       end
       it 'removes VM from all server pools' do
-        simple_vm_lifecycle(cpi, @nsxt_opaque_vlan_1, vm_type) do |vm_id|
-          vm = cpi.vm_provider.find(vm_id)
-          vm_ip = vm.mob.guest&.ip_address
-          expect(vm_ip).to_not be_nil
+        if use_policy_api?
```

Comment:
instead of putting these `if ... else ... end` in each of the tests can we split the relevant tests into 2; one tagged as a manager test and the other tagged as a policy test. It feels like it would be cleaner and in the future make it easier to delete all the old manager api tests.

If need be, we can also create separate files entirely. Im not sure which would be best here.

(same for all the tests in this file)

(also im impressed; this seems like a ton of work; nice job)